test/long/org/apache/cassandra/cql3/CorruptionTest.java

```diff
@@ -59,7 +60,11 @@ public class CorruptionTest extends SchemaLoader
     @BeforeClass()
     public static void setup() throws ConfigurationException, IOException
     {
-        Schema.instance.clear();
+        DatabaseDescriptor.daemonInitialization();
+
+        // clean-up is only needed if we run the test from IDE
+        ServerTestUtils.mkdirs();
+        ServerTestUtils.cleanup();
```

Comment:
This is always done before any instantiation of `EmbeddedCassandraService`, so we may make tests slightly briefer and less error prone if we encapsulate both things into a single utility method. Going a bit further, we could move the entire `EmbeddedCassandraService` class into the tests directory (it's only used for  tests), and place the cleanup of dirs into its constructor. wdyt?

Reply:
I was thinking about that, I agree we can safely have a single method which includes `mkdirs` and `cleanup`., or even have a static methods which does that and creates ECS. However, I'm not sure we should move `EmbeddedCassandraService` to the tests directory because it is currently a user visible thing and some users may start Cassandra that way. When it was moved into the tests directory, it would not be a part of the `cassandra-all` artifact in public Maven repository. This is also why we should not put `mkdirs` and `cleanup` into the constructor

Reply:
Right, there could be someone using `EmbeddedCassandraService` from the Maven artifact. Then we can put the repeated block of code into a test method into a utility method, more or less [this way](https://github.com/adelapena/cassandra/commit/c6f25e16f548c929bc79e6e91d9efc8df6ad3124). I think that would save us some code duplication, wdyt?

Reply:
I agree with everything said here, maybe I would just rename the method as we already have start in `EmbeddedCassandraService` itself